limit.go

```diff
@@ -0,0 +1,75 @@
+package pilosa
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"time"
+)
+
+var (
+	ErrorTooManyConnections = errTooManyConnections{}
+)
+
+func NewBoundListener(maxActive int, l net.Listener) net.Listener {
+	b := &boundListener{l, make(chan struct{}, maxActive)}
+	for i := 0; i < maxActive; i++ {
+		b.active <- struct{}{}
+	}
+	return b
+}
+
+type boundListener struct {
+	net.Listener
+	active chan struct{}
+}
+
+type boundConn struct {
+	net.Conn
+	active chan struct{}
+}
+type errTooManyConnections struct {
+}
+
+func (e errTooManyConnections) Error() string   { return "To Many Connections" }
+func (e errTooManyConnections) Timeout() bool   { return false }
+func (e errTooManyConnections) Temporary() bool { return true }
+
+func (l *boundListener) Accept() (net.Conn, error) {
+	select {
+	case <-l.active:
+		c, err := l.Listener.Accept()
+		if err != nil {
+			l.active <- struct{}{}
+			return nil, err
+		}
+		return &boundConn{c, l.active}, err
+	default: //out of connections
+		c, _ := l.Listener.Accept()
+		l.writeError(c, "Too Many Connections")
```

Comment:
can you use `ErrorTooManyConnections.Error()` here instead of repeating the string?